kura/org.eclipse.kura.core.crypto/src/main/java/org/eclipse/kura/core/crypto/CryptoServiceImpl.java

```diff
@@ -92,73 +93,19 @@ protected void activate() {
     }
 
     private byte[] base64Decode(String internalStringValue) {
-        Object convertedData = null;
-        try {
-            Class<?> clazz = Class.forName("javax.xml.bind.DatatypeConverter");
-            Method method = clazz.getMethod("parseBase64Binary", String.class);
-            convertedData = method.invoke(null, internalStringValue);
-        } catch (ClassNotFoundException e) {
-            convertedData = base64DecodeJava8(internalStringValue);
-        } catch (LinkageError e) {
-            convertedData = base64DecodeJava8(internalStringValue);
-        } catch (Exception e) {
-
-        }
-
-        if (convertedData != null) {
-            return (byte[]) convertedData;
-        }
-        return null;
-    }
-
-    private Object base64DecodeJava8(String internalStringValue) {
-        Object convertedData = null;
-        try {
-            Class<?> clazz = Class.forName("java.util.Base64");
-            Method decoderMethod = clazz.getMethod("getDecoder", (Class<?>[]) null);
-            Object decoder = decoderMethod.invoke(null, new Object[0]);
-
-            Class<?> base64Decoder = Class.forName("java.util.Base64$Decoder");
-            Method decodeMethod = base64Decoder.getMethod("decode", String.class);
-            convertedData = decodeMethod.invoke(decoder, internalStringValue);
-        } catch (Exception e1) {
-        }
-        return convertedData;
+        return Base64.decodeBase64(internalStringValue);
     }
 
     private String base64Encode(byte[] encryptedBytes) {
-        Object convertedData = null;
         try {
-            Class<?> clazz = Class.forName("javax.xml.bind.DatatypeConverter");
-            Method method = clazz.getMethod("printBase64Binary", byte[].class);
-            convertedData = method.invoke(null, encryptedBytes);
-        } catch (ClassNotFoundException e) {
-            convertedData = base64EncodeJava8(encryptedBytes);
-        } catch (LinkageError e) {
-            convertedData = base64EncodeJava8(encryptedBytes);
-        } catch (Exception e) {
-
+            /* This try-catch would not be necessary if we would have a
+             * more recent version of commons.net of Java 7.
+             * But UTF-8 is always supported with Java.
+             */
+            return new String(Base64.encodeBase64(encryptedBytes, false), "UTF-8");
+        } catch (UnsupportedEncodingException e) {
+            throw new RuntimeException(e);
```

Comment:
I don't see a big difference, the exception will never ever be thrown and the whole try-catch will be gone with java 7 anyway.